core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -782,6 +782,13 @@ public String toString() {
               + " be used.", Name.CONF_DIR))
           .setScope(Scope.ALL)
           .build();
+  public static final PropertyKey ROCKS_GRACEFUL_SHUTDOWN_TIMEOUT =
```

Comment:
i guess based on our discussion we don't actually need this correct?

Reply:
yes if in our review we think the flag + RWLock approach is handling all race conditions, I will remove the grace period and all relevant code

Reply:
So I think you can keep this, but just wait for a maximum amount of time, and if you can't shut down within this time, you log a warning and say that we are shutting down unsafely because a thread still has a lock on the object.

Basically if we see this warning log, we know something is not working correctly that we need to debug.